ext/msgpack/unpacker.c

```diff
@@ -583,7 +525,7 @@ static int read_primitive(msgpack_unpacker_t* uk)
                 READ_CAST_BLOCK_OR_RETURN_EOF(cb, uk, 4);
                 uint32_t count = _msgpack_be32(cb->u32);
                 if(count == 0) {
-                    return object_complete_string(uk, rb_str_buf_new(0));
+                    return object_complete(uk, rb_utf8_str_new_static("", 0));
```

Comment:
Two things here.

Since I removed `object_complete_string` and `object_complete_binary` we have to use API that set the proper encoding directly.

And also `rb_str_buf_new` is meant for string that will be concatenated to, so they are instantiated with a capacity of 63 bytes (so 103bytes total):

```
#define STR_BUF_MIN_SIZE 63
```

Since here we only want an empty string, it's a bit of a waste.

Reply:
I may be misunderstanding your comment. Do you mean:
* the previous code using `rb_str_buf_new` instantiated 63 bytes larger value
* the newer code will not consume additional 63 bytes
Is that correct?

Reply:
The first one. `rb_str_buf_new(0)` will always instantiate a string with 63 bytes of capacity, so a total of 103 bytes.

The new code will directly right size the string, so saving up to 63 bytes.